### PR TITLE
Adds core style overrides & bumps jetpack version

### PR DESCRIPTION
**Pull request by cpapazoglou**
This is a follow up of https://github.com/Automattic/jetpack/pull/18097. Related issue is https://github.com/Automattic/jetpack/issues/18092

#### Changes proposed in this Pull Request:

* Bumps jetpack version as a cache buster solution for styles loaded from wpcom (discussed here p1608286346170200-slack-CBG1CP4EN)
* Restructures core styles overrides to be compiled by a template
* Conditionally loads the correct styles override file
* Adds class to site-switcher and replaces `#adminmenuwrap > #adminmenu > .wp-first-item.menu-top-first > .wp-first-item.menu-top-first ` with `#adminmenuwrap > #adminmenu .site-switcher `
* Fixes Site-Switcher and Notes states

### **Simple Sites**
1) Enable nav-unification experiment (see paYJgx-1af-p2)
2) Apply the patch D54531
3) Change color-scheme from `https://wordpress.com/me/account` to a core color scheme (see list below)

### **Atomic Sites**
1) Select this branch on jetpack on an Atomic site (can't find it right now though. )
2) Install nav-unification enabler (needs to be proxied)
[nav-unification.zip](https://github.com/Automattic/jetpack/files/5717222/nav-unification.zip)
3) Change color-scheme from `/wp-admin/profile.php`

**Check the following color-schemes:**
- default
- light
-  Modern
-  Blue
-  Coffee
-  Ectoplasm
-  Midnight
-  Ocean
-  Sunrise

for the following:
**Masterbar:**
- [x] Masterbar background color
- [x] Masterbar hover color (Hover on Reader)
- [x] Masterbar active color (My Sites)

**Sidebar:**
- [x] Site card fadeout
- [x] Site card background color is not changing on hover
- [x] Site card icon color

#### Proposed changelog entry for your changes:

* Restructures color-scheme styles for nav-unification

### CODE REVIEW

**Review by frontdevde — COMMENTED** [review]
First of all kudos for making the effort to switch the plain css over to generating the overrides per scheme via scss. This will significantly reduce the maintenance burden in the future while also making it less error prone.

> Select this branch on jetpack on an Atomic site (can’t find it right now though. )

Same here, the branch name doesn't show on the Jetpack beta plugin.

For Simple sites I applied the diff and noticed that it couldn't be cleanly applied which lead to the addition of  an `.orig` file on the diff. Seems be an svn issue. Let's make sure the php file is as expected on the diff and then remove the `.orig` file. 

While testing for Simple sites, the override was loaded in addition to the scheme as expected:

Screenshot 2020-12-21 at 08 24 06

On Simple sites, I noticed that the active state for the Notification bell in the wpcom masterbar isn't yet accounted for. Let's add that:

|Calypso|wp-admin|
|-|-|
|Screenshot 2020-12-21 at 08 26 54|Screenshot 2020-12-21 at 08 26 22|

Edit: 

Same goes for the hover state on the Notification bell. 
See the relevant selectors here https://github.com/Automattic/jetpack/pull/18115/files#diff-bf5e4657a1c54509b24846ff67f30e788493be6b9d7b34cddc6575225ba9d84eR73-R81

File: modules/masterbar/admin-color-schemes/colors/_core-overrides.scss [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+#wpadminbar:not(.mobile) .ab-top-menu > li:hover > .ab-item,
+#wpadminbar #wp-admin-bar-notes.hover > .ab-item {
+	background: $menu-submenu-background;
+}
+
+/**
+ * Site Card
+ */
+ #adminmenu li.toplevel_page_site-card {
```
Comment (frontdevde) [inline modules/masterbar/admin-color-schemes/colors/_core-overrides.scss:25]: Looks like there's unnecessary whitespace at the beginning of the line here

Comment (cpapazoglou) [reply modules/masterbar/admin-color-schemes/colors/_core-overrides.scss]: Removed 40d2b27

**Comment by cpapazoglou** [comment]
> First of all kudos for making the effort to switch the plain css over to generating the overrides per scheme via scss. This will significantly reduce the maintenance burden in the future while also making it less error prone.
> 
> > Select this branch on jetpack on an Atomic site (can’t find it right now though. )
> 
> Same here, the branch name doesn't show on the Jetpack beta plugin.
> 
> For Simple sites I applied the diff and noticed that it couldn't be cleanly applied which lead to the addition of an `.orig` file on the diff. Seems be an svn issue. Let's make sure the php file is as expected on the diff and then remove the `.orig` file.
> 
> While testing for Simple sites, the override was loaded in addition to the scheme as expected:
> 
> Screenshot 2020-12-21 at 08 24 06
> 
> On Simple sites, I noticed that the active state for the Notification bell in the wpcom masterbar isn't yet accounted for. Let's add that:
> 
> Calypso	wp-admin
> Screenshot 2020-12-21 at 08 26 54	Screenshot 2020-12-21 at 08 26 22
> Edit:
> 
> Same goes for the hover state on the Notification bell.
> See the relevant selectors here https://github.com/Automattic/jetpack/pull/18115/files#diff-bf5e4657a1c54509b24846ff67f30e788493be6b9d7b34cddc6575225ba9d84eR73-R81

Thanks @frontdevde for wrangling this testing! I have addressed your feedback!

**Review by frontdevde — COMMENTED** [review]
I just noticed another very subtle issue when retesting with `blue`. The Site Switcher text has an alternative text color in nav unification that is not yet accounted for:

|Calypso|wp-admin|
|-|-|
|Screenshot 2020-12-21 at 11 18 16|Screenshot 2020-12-21 at 11 18 26|

For the color schemes originating in Calypso this was addressed here: 
https://github.com/Automattic/jetpack/pull/18115/files#diff-bf5e4657a1c54509b24846ff67f30e788493be6b9d7b34cddc6575225ba9d84eR116-R130

File: modules/masterbar/admin-color-schemes/colors/_overrides.scss [hunk 2]
```diff
@@ -114,14 +114,14 @@
 	}
 
 	// Site Card - Browse Sites button text & arrow
-	#adminmenuwrap > #adminmenu > .wp-first-item.menu-top-first > .wp-first-item.menu-top-first {
+	#adminmenuwrap > #adminmenu .site-switcher {
```
Comment (cpapazoglou) [inline modules/masterbar/admin-color-schemes/colors/_overrides.scss:117]: There are cases when "Browse Sites" is not rendered and thus not the first item

**Review by frontdevde — COMMENTED** [review]
Thanks for fixing the regressions as well! 🙌

Starting to look really solid! I've re-tested all color schemes again on Simple and the only issues I found were the following:

- On the Light color scheme the Notification bell  icon is white when it's black in Calypso.

- Related, on the Light color scheme the active state of the Notification bell is still off:

|Calypso|wp-admin|
|-|-|
|Screenshot 2020-12-21 at 12 37 43|Screenshot 2020-12-21 at 12 38 01|

- In [Coffee, Blue, Ectoplasm, Midnight, Ocean, Sunrise] the Write button looks off on hover:

|Calypso|wp-admin|
|-|-|
|Screenshot 2020-12-21 at 12 40 32|Screenshot 2020-12-21 at 12 40 23|

**Comment by cpapazoglou** [comment]
Fixed both issues and refreshed the patch

**Comment by frontdevde** [comment]
Looking good in all color schemes on Simple now 👏

Now that the branch is available in Jetpack Beta, I started testing as per your Atomic instructions and I'm noticing that the box-shadow at the bottom of the Masterbar differs:

|Calypso & Simple|Atomic|
|-|-|
|Screenshot 2020-12-21 at 13 35 08|Screenshot 2020-12-21 at 13 35 27|

For the Calypso schemes this was addressed here https://github.com/Automattic/jetpack/pull/18115/files#diff-bf5e4657a1c54509b24846ff67f30e788493be6b9d7b34cddc6575225ba9d84eR62-R65

File: jetpack.php [hunk 3]
```diff
@@ -16,7 +16,7 @@
 
 define( 'JETPACK__MINIMUM_WP_VERSION', '5.5' );
 define( 'JETPACK__MINIMUM_PHP_VERSION', '5.6' );
-define( 'JETPACK__VERSION', '9.3-alpha' );
+define( 'JETPACK__VERSION', '9.3-alpha-20201221' );
```
Comment (frontdevde) [inline jetpack.php:19]: Was this for testing purposes or were you intending for this to go in?

**Review by frontdevde — APPROVED** [review]
Thanks for addressing all the feedback, looking slick now!

Tested as per testing instructions and can confirm it's now looking as expected in both Simple and WoA.

I'd appreciate if you could update the PR description with the latest changes so it provides some context for the next person reviewing. That said, I think we can have Crew take a look at this.

File: jetpack.php (hunk 3 above)
Comment (cpapazoglou) [reply jetpack.php:19]: This was intentional in order to bust the caches, discussed here p1608286346170200-slack-CBG1CP4EN

**Review by leogermani — APPROVED** [review]
@zinigor and I reviewed this PR and are approving it.

We found one issue in the WP-admin for atomic sites: when switching between color schemes, the master bar is not changing until a full page refresh.

This needs more testing and we aim to make it easier for testing approving this PR with a few usability issues still present.

slack discussion: p1608616675330200-slack-CBG1CP4EN

**Comment by cpapazoglou** [comment]
Deployed r218681-wpcom
